### PR TITLE
added pytest to conda-forge build

### PR DESCRIPTION
**Pull request by AndrewAnnex**
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Adds pytest unit tests to build process to help solve future issues like https://github.com/AndrewAnnex/SpiceyPy/issues/305

### CODE REVIEW

**Comment by AndrewAnnex** [comment]
okay.. the patch is correct but because we are grabbing the source from pypi then we run into a problem as the archive does not contain the tests. I might have to issue a new release with the current pr https://github.com/AndrewAnnex/SpiceyPy/pull/306 merged in

File: recipe/meta.yaml [hunk 1]
```diff
@@ -31,9 +32,10 @@ requirements:
     - six
 
 test:
-  imports:
-    - spiceypy
-    - spiceypy.utils
+  requires:
+    - pytest
+  commands:
+    - py.test --pyargs spiceypy
```
Comment (Juanlu001) [inline recipe/meta.yaml]: Only one thing, use `pytest` instead of `py.test`!

Comment (AndrewAnnex) [reply recipe/meta.yaml]: done, but not sure that will fix it...

**Comment by AndrewAnnex** [comment]
Forgot that I needed the test source to be included in the pypi source code. This PR is blocked until https://github.com/AndrewAnnex/SpiceyPy/pull/306 is merged

**Comment by AndrewAnnex** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml (hunk 1 above)
Comment (Juanlu001) [reply recipe/meta.yaml]: Yeah, it was not to fix it, but to use the "new" command

**Comment by AndrewAnnex** [comment]
okay now some progress, looks like this is where things start to fail:
```
2019-10-20T00:26:46.1931918Z spiceypy/tests/test_wrapper.py::test_bltfrm PASSED                       [  6%]
2019-10-20T00:26:46.9316354Z spiceypy/tests/test_wrapper.py::test_bodc2n FAILED                       [  6%]
2019-10-20T00:26:47.6349141Z spiceypy/tests/test_wrapper.py::test_bodc2s FAILED                       [  6%]
2019-10-20T00:26:48.5494272Z spiceypy/tests/test_wrapper.py::test_boddef PASSED                       [  6%]
```
so it has something to do with the downloaded kernels/furnsh loading them

**Comment by AndrewAnnex** [comment]
excellent, I will do two things next, make sure to use a cross-platform compatible version of the patch code and commit it to the master repo, will mirror those changes in the patch. 

it would be great to also run tests in this repo on windows, which will currently fail until https://github.com/conda-forge/cspice-feedstock/pull/12 pr fixes are completed

**Comment by AndrewAnnex** [comment]
@conda-forge-admin, please rerender

**Comment by Juanlu001** [comment]
:tada:

**Pull request merged**